x-pack/qa/multi-node/src/javaRestTest/java/org/elasticsearch/multi_node/GlobalCheckpointSyncActionIT.java

```diff
@@ -12,14 +12,28 @@
 import org.elasticsearch.common.settings.SecureString;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.util.resource.TextResource;
 import org.elasticsearch.test.rest.ESRestTestCase;
 import org.elasticsearch.test.rest.ObjectPath;
 import org.elasticsearch.xcontent.XContentBuilder;
+import org.junit.ClassRule;
 
 import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
 import static org.hamcrest.Matchers.equalTo;
 
 public class GlobalCheckpointSyncActionIT extends ESRestTestCase {
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
```

Comment:
Just to double check, we can have these class rules on a test superclass right? That is, I hope we do not have to repeat cluster configuration on every concrete class, but can share it through a common test case. For example, we have lots of abstract test cases in xpack security that it would be a real pain to have to duplicate all the configuration.

Reply:
Yes, you can define a class rule on the superclass and it'll apply to subclasses. These things are immutable though and cannot be overriden (sicne it's `static` it belongs to the parent class) so that's a workable scenario only when you know the config is indeed identical.

The other option if that's not entirely possible (each subclass needs some slightly different config) is to put the common config in a `LocalClusterConfigProvider` and do `ElasticsearchCluster.local().apply(commonConfig)`.

Reply:
Thinking about this more I'll have to test the parent class situation. Since these are `static` members that means the cluster itself would be shared across child classes which is probably ok since only one would be excuting at one time in any given test worker. I'll need to explicitly test this.

Reply:
Actually yes, this should be absolutely fine since specs are immutable (on purpose) and we create a new cluster factory inside the rule `apply()` method so each test case will actually get its own cluster.